src/components/StoreList/StoreList.js

```diff
@@ -29,17 +17,49 @@ export default class StoreList extends React.Component {
 					items={this.state.storeList}
 					onClick={(storeCode, itemName) => {
 						try {
-							sendSlackMessage(
-								this.props.username,
-								itemName,
-								storeCode
+							let id = getUserSlackID(
+								this.props.currentUser,
+								this.props.users
 							);
+							if (!id) throw new Error();
+							sendSlackMessage(id, itemName, storeCode);
 						} catch (error) {
-							alert(error);
+							this.setState({sendSlackMessageError: true});
 						}
 					}}
 				/>
+				{this.state.sendSlackMessageError && (
+					<ErrorMessage text={'failed to send Slack message'} />
+				)}
 			</div>
 		);
 	}
 }
+
+const mapStateToProps = state => {
+	return {
+		storeList: state.storeList
+	};
+};
+
+const mapDispatchToProps = dispatch => {
+	return {
+		setStore: getStore((err, items) => {
+			if (err) return;
+			let storeList = items.map(item => ({
+				name: item.name + (item.qualifier ? ' ' + item.qualifier : ''),
+				index: item.id
+			}));
+			dispatch({
```

Comment:
We could extract this to it's own action to help with typos, could be worth renaming as it can be confusing which store we're talking about - honesty or redux!